tgui/packages/tgui/interfaces/KitchenMachine.js

```diff
@@ -0,0 +1,103 @@
+import { useBackend } from '../backend';
+import { Button, Section, Table, Flex, Icon, Dimmer } from '../components';
+import { Window } from '../layouts';
+
+export const KitchenMachine = (props, context) => {
+  const { data } = useBackend(context);
+  const { ingredients } = data;
+
+  return (
+    <Window resizable>
+      <Window.Content scrollable display="flex" className="Layout__content--flexColumn">
+        <Operating/>
+        <KitchenTop/>
+        <Section title="Ingredients">
+          <Table className="Ingredient__Table">
+            {ingredients.map((product) => (
+              <Table.Row tr={5} key={product.name}>
+                <tr>
+                  <td>
+                    <Table.Cell bold>{product.name}</Table.Cell>
+                  </td>
+                </tr>
+                <td>
+                  <Table.Cell collapsing textAlign="center">
+                    {product.amount} {product.units}
+                  </Table.Cell>
+                </td>
+              </Table.Row>
+            ))}
+          </Table>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const KitchenTop = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+          inactive,
+          tooltip
+        } = data;
+
+  return (
+    <Section title="Controls">
+      <Flex>
+        <Flex.Item width="50%" mr="3px">
+          <Button
+            fluid
+            textAlign="center"
+            icon="power-off"
+            disabled={inactive}
+            tooltip={tooltip}
+            tooltipPosition="bottom"
+            content="Activate"
+            onClick={() => act('cook')}
+          />
+        </Flex.Item>
+        <Flex.Item width="50%" mr="3px">
```

Comment:
There slightly too much padding on the right with this
```suggestion
        <Flex.Item width="50%">
```